src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -41,6 +41,11 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
   private videoMaxBandwidthKbps: number = DefaultDeviceController.defaultVideoMaxBandwidthKbps;
 
   private useWebAudio: boolean = false;
+  private deviceIdGroupIdMap: { [mediaDeviceKind: string]: Map<Device, string> } = {
```

Comment:
Few browsers like Safari does not have groupId as part of the device information. To make the solution oblivious of the browser I added this state so that it can have a mapping between the deviceId and the groupId (or '' if group id is missing)

Reply:
Do the browsers that have an empty group ID require the workaround logic?

Reply:
No browsers such as Safari does not have groupId. It has unique deviceId which is never reassigned like in the case of chromium based browser.